pool.go

```diff
@@ -86,6 +93,14 @@ func PoolConnFunc(cf ConnFunc) PoolOpt {
 	}
 }
 
+// PoolDialTimeout specifies the timeout for pool initialization.
+// NewPool will fail if connections are not created after the deadline.
+func PoolDialTimeout(d time.Duration) PoolOpt {
```

Comment:
The function name will probably confuse people since this is not about dialing redis but filling the pool.